Objects/bytes_methods.c

```diff
@@ -557,10 +557,52 @@ find_internal(const char *str, Py_ssize_t len,
 }
 
 PyObject *
-_Py_bytes_find(const char *str, Py_ssize_t len, PyObject *sub,
+_Py_bytes_find(const char *str, Py_ssize_t len, PyObject *subobj,
                Py_ssize_t start, Py_ssize_t end)
 {
-    Py_ssize_t result = find_internal(str, len, "find", sub, start, end, +1);
+    Py_ssize_t result;
+    if (PyTuple_Check(subobj)) {
+        result = -1;
+        ADJUST_INDICES(start, end, len);
+        // Work in batches of 10000
```

Comment:
Could this split words? For example, `name = string[9995:10005]`

Reply:
No, it searches in `string[0:10000 + len(substr)]`.

Reply:
Any test case addressing this?

Reply:
Not yet, I'll add dgrigonis' suggestion: https://github.com/python/cpython/pull/119501#discussion_r1614098283